src/clr-angular/data/tree-view/models/selected-state.ts

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+// TODO: I'd like this to be a CheckedState enum for the checkboxes in the future.
+export enum ClrSelectedState {
```

Comment:
Unless we are expecting external use of this enum, we should name it `SelectedState`. I also suggest `CheckboxSelectedState` or something a little more precise.

Reply:
- It needs to be public, since we're going to offer a `[(clrSelected)]` two-way binding accepting and emitting this enum. This replaces the double two-way binding of `[(clrSelected)]` and `[(clrIndeterminate)]`.
- It is not tied to checkbox, it just describes a selection state. It turns out our implementation uses checkboxes, but Until we move checkboxes to this pattern, I'd like to keep the name abstract in that it just describes if a consumer's object is selected or not. And it's used in Tree here, but it could make sense in combobox or complex list components.